src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -48,147 +46,95 @@
-    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(myCapital, 3, player);
+    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(ProData.myCapital, 3, player);
     ProLogger.debug("Currently in defensive stance: " + isDefensive);
 
-    // Find all purchase options and min cost per hit point
-    final List<ProPurchaseOption> specialPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> factoryPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> landPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> airPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> seaPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    ProPurchaseUtils.findPurchaseOptions(player, landPurchaseOptions, airPurchaseOptions, seaPurchaseOptions,
-        factoryPurchaseOptions, specialPurchaseOptions);
-    minCostPerHitPoint = ProPurchaseUtils.getMinCostPerHitPoint(player, landPurchaseOptions);
-
     // Find the maximum number of units that can attack each territory and max enemy defenders
-    final List<Territory> myUnitTerritories =
-        Match.getMatches(data.getMap().getTerritories(), Matches.territoryHasUnitsOwnedBy(player));
-    attackOptionsUtils.findAttackOptions(player, myUnitTerritories, attackMap, unitAttackMap, transportAttackMap,
-        bombardMap, transportMapList, new ArrayList<Territory>(), new ArrayList<Territory>(),
-        new ArrayList<Territory>(), false, false);
-    attackOptionsUtils.findScrambleOptions(player, attackMap);
-    final ProMoveOptions alliedAttackOptions = attackOptionsUtils.findAlliedAttackOptions(player);
-    final ProMoveOptions enemyDefendOptions = attackOptionsUtils.findEnemyDefendOptions(player);
+    territoryManager.populateAttackOptions();
+    territoryManager.populateEnemyDefenseOptions();
 
     // Remove territories that aren't worth attacking and prioritize the remaining ones
-    final List<ProTerritory> prioritizedTerritories =
-        attackOptionsUtils.removeTerritoriesThatCantBeConquered(player, attackMap, unitAttackMap, transportAttackMap,
-            alliedAttackOptions, enemyDefendOptions, false);
-    List<Territory> territoriesToAttack = new ArrayList<Territory>();
-    for (final ProTerritory patd : prioritizedTerritories) {
-      territoriesToAttack.add(patd.getTerritory());
+    final List<ProTerritory> attackOptions = territoryManager.removeTerritoriesThatCantBeConquered();
+    List<Territory> clearedTerritories = new ArrayList<Territory>();
+    for (final ProTerritory patd : attackOptions) {
+      clearedTerritories.add(patd.getTerritory());
     }
-    ProMoveOptions enemyAttackOptions =
-        attackOptionsUtils.findEnemyAttackOptions(player, territoriesToAttack, new ArrayList<Territory>());
+    territoryManager.populateEnemyAttackOptions(clearedTerritories, new ArrayList<Territory>());
```

Comment:
Perhaps return an `Optional<ArrayList>` if you want an output value, otherwise you have two methods, one signature that takes an array list, and a second that has one few parameters.